developer-docs/tips-for-bazel.md

```diff
@@ -0,0 +1,95 @@
+# Bazel Tips
```

Comment:
Can we link to this from `SUMMARY.md`? Not sure where, maybe under Monorepo or Contribute...